script.module.resolveurl/lib/resolveurl/plugins/solidfiles.py

```diff
@@ -0,0 +1,35 @@
+"""
+    Plugin for ResolveUrl
+    Copyright (C) 2022 shellc0de
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from resolveurl.lib import helpers
+from resolveurl.plugins.__resolve_generic__ import ResolveGeneric
+
+
+class SolidFilesResolver(ResolveGeneric):
+    name = 'SolidFiles'
+    domains = ['solidfiles.com']
+    pattern = r'(?://|\.)(solidfiles\.com)/(?:e|v)/([0-9a-zA-Z]+)'
+
+    def get_media_url(self, host, media_id):
+        return helpers.get_media_url(self.get_url(host, media_id),
+                                     patterns=[r'''downloadUrl":"(?P<url>[^"]+)'''],
+                                     generic_patterns=False,
+                                     result_blacklist=['.zip', '.rar'])
```

Comment:
Add '.7z' as well please into the black_list

Reply:
Done :)